Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs

```diff
@@ -305,7 +324,21 @@ private static float GetAmbientPostArmorDamage(float dmgAmount, StatDef armorRat
                     }
                 }
             }
-            return dmgAmount * dmgMult;
+            
+            
+              if (pawn.TryGetComp<Comp_BurnDamageCalc>() != null)
+              {
+                if (armorRatingStat == StatDefOf.ArmorRating_Heat)
+                {
+                    if (pawn.TryGetComp<Comp_BurnDamageCalc>().deflectedSharp)
+                    {
+                            dmgMult /= 2f;
+                    }
+                }
+                pawn.TryGetComp<Comp_BurnDamageCalc>().deflectedSharp = false;

```

Comment:
Why are you using `Comp_BurnDamageCalc` to track `deflectedSharp`, instead of adding a `deflectedSharp` field to the `DamageInfo dinfo`?